yaci-indexer/src/main/java/org/cardanofoundation/rosetta/yaciindexer/stores/txsize/CustomTransactionSizeStore.java

```diff
@@ -0,0 +1,229 @@
+      transactionSizeEntities.add(transactionSizeEntity);
+    });
+
+    if(!transactionSizeEntities.isEmpty()) {
+      transactionSizeRepository.saveAll(transactionSizeEntities);
+    }
+  }
+
+  /**
+   * Reconstructing the Witness_set from transaction and adding it to signedTransaction
+   * CDDL Definition: https://github.com/IntersectMBO/cardano-ledger/blob/e6b6d4f85fb72b5cb5b5361e534d3bb71bb9e55e/eras/alonzo/impl/cddl-files/alonzo.cddl#L269
+   * @param tx
+   * @param signedTransaction
+   * @return
+   */
+  private static int addWitnessSetToSignedTransaction(Transaction tx, Map signedTransaction) {
+    // adding witnesses to signedTransaction
+    Map witnessSet = new Map();
+    int scriptSize = 0;
+    addvKeyWitnessToWitness(tx, witnessSet);
+    addNativeScriptsToWitness(tx, witnessSet);
+    addBootstrapToWitness(tx, witnessSet);
+
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV1Scripts(),
+        TransactionBuildingConstants.PLUTUSV1_WITNESS_INDEX);
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV2Scripts(), TransactionBuildingConstants.PLUTUSV2_WITNESS_INDEX);
+    scriptSize += addPlutusToWitness(witnessSet, tx.getWitnesses().getPlutusV3Scripts(), TransactionBuildingConstants.PLUTUSV3_WITNESS_INDEX);
+
+    addDatumToWitness(tx, witnessSet);
+    addRedeemerToWitness(tx, witnessSet);
+
+    if(!witnessSet.getKeys().isEmpty()) {
+      signedTransaction.put(new UnsignedInteger(TransactionBuildingConstants.WITNESS_SET_INDEX), witnessSet);
+    }
+    return scriptSize;
+  }
+
+  /**
+   * Adding the datum to Witness set. CDDL spec: https://github.com/IntersectMBO/cardano-ledger/blob/e6b6d4f85fb72b5cb5b5361e534d3bb71bb9e55e/eras/alonzo/impl/cddl-files/alonzo.cddl#L280
+   * @param tx transaction to extract the datum
+   * @param witnessSet witnessSet to add the datum
+   */
+  private static void addDatumToWitness(Transaction tx, Map witnessSet) {
+    if(!tx.getWitnesses().getDatums().isEmpty()) {
+      Array array = new Array();
+      // could speed it up by passing an empty array, since we are only interested in the size not the content
+      tx.getWitnesses().getDatums().forEach(datum -> array.add(new ByteString(HexUtil.decodeHexString(datum.getCbor()))));
+      witnessSet.put(new UnsignedInteger(TransactionBuildingConstants.PLUTUS_DATUM_WITNESS_INDEX), array);
+    }
+  }
+
+  /**
+   * Adding Plutus Script data to witnessSet. Can be used for V1, V2 and V3. CDDL spec: https://github.com/IntersectMBO/cardano-ledger/blob/e6b6d4f85fb72b5cb5b5361e534d3bb71bb9e55e/eras/alonzo/impl/cddl-files/alonzo.cddl#L278
+   * @param witnessSet witnessset to add the data
+   * @param scripts List of PlutusScripts
+   * @param witnessSetIndex Index where to add the datum based on cddl spec
+   * @return
+   */
+  private static int addPlutusToWitness(Map witnessSet, List<PlutusScript> scripts, int witnessSetIndex) {
+    AtomicInteger scriptSize = new AtomicInteger();
```

Comment:
Will we introduce in this method multithreading? If the script list is big enough, it will be beneficial. Or there is something different behind this AtomicInteger?